### PR TITLE
Please review: S-matrix inclusion to the electron dynamics failed

### PR DESCRIPTION
**Pull request by Pulokesh**

### CODE REVIEW

File: src/md.py [hunk 1]
```diff
@@ -183,6 +183,8 @@ def run_MD(syst,el,ao,E,sd_basis,params,label,Q, active_space):
     eext = [0.0]*ens_sz
     mu = []
     smat = CMATRIX(nstates,nstates)
+    for i in xrange(nstates):
+        smat.set(i,i,1.0,0.0)
```
Comment (Pulokesh) [inline src/md.py]: S matrix is initialized such that diagonal elements are 1.0 (orthonormal basis)

File: src/x_to_libra_qe.py [hunk 2]
```diff
@@ -184,6 +184,7 @@ def qe_to_libra(params, E, sd_basis, label, mol, suff, active_space):
     # The names "E_mol" and "D_mol" are confusing when you have already constructed Energy and NACs           
     # based on SD basis sets.
     # ********************************************************************************************
+    S_mol = average_S(P11,P22)  # Average S(t+dt/2) = (S(t) + S(t+dt))/2.0
```
Comment (alexvakimov) [inline src/x_to_libra_qe.py]: actually, I doubt we need a special function for computing an average of two matrices..

File: src/x_to_libra_qe.py [hunk 3]
```diff
@@ -184,6 +184,7 @@ def qe_to_libra(params, E, sd_basis, label, mol, suff, active_space):
...
     E_mol = average_E(E,E2)
```
Comment (alexvakimov) [inline src/x_to_libra_qe.py]: same is here

File: src/x_to_libra_qe.py [hunk 4]
```diff
@@ -184,6 +184,7 @@ def qe_to_libra(params, E, sd_basis, label, mol, suff, active_space):
...
     D_mol = NAC(P12,P21,params["dt_nucl"])
```
Comment (alexvakimov) [inline src/x_to_libra_qe.py]: again, can simplify this to be a one-liner rather than a call of a function

File: src/md.py [hunk 5]
```diff
@@ -208,7 +209,8 @@ def run_MD(syst,el,ao,E,sd_basis,params,label,Q, active_space):
 
                         # Electronic propagation: half-step
                         for k in xrange(el_mts):
-                            el[cnt].propagate_electronic(0.5*dt_elec, ham[cnt])
+                            #el[cnt].propagate_electronic(0.5*dt_elec, ham[cnt])
+                            el[cnt].propagate_electronic(0.5*dt_elec, ham[cnt], smat)  # el propagate using S-matrix
```
Comment (alexvakimov) [inline src/md.py]: please, make the choice of either method an option, controlled by an external parameter

File: src/md.py (hunk 1 above)
Comment (alexvakimov) [inline src/md.py]: of course

File: src/x_to_libra_qe.py (hunk 2 above)
Comment (Pulokesh) [inline src/x_to_libra_qe.py]: Yes, I agree.

File: src/x_to_libra_qe.py (hunk 3 above)
Comment (Pulokesh) [inline src/x_to_libra_qe.py]: This is the same line.

File: src/x_to_libra_qe.py (hunk 4 above)
Comment (Pulokesh) [inline src/x_to_libra_qe.py]: Same line. S_mol = 0.50*(P11+P22) should work.

**Pull request merged**